app/webpacker/controllers/turbo_search_controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ['form'];
+
+  connect() {
+    this.formTarget.addEventListener('submit', this.turboSubmit);
+  }
+
+  disconnect() {
+    this.formTarget.removeEventListener('submit', this.turboSubmit);
+  }
+
+  turboSubmit(e) {
+    e.preventDefault();
+    const form = e.target;
+    const url = new URL(form.action);
+    const formData = new FormData(form);
+    const params = new URLSearchParams(formData).toString();
+    
+    // Manually visit the new URL with the search params
+    // inorder to preserve params
+    Turbo.visit(`${url.pathname}?${params}`, { action: 'replace' });
+  }
+}
```

Comment:
Nice work, but this looks like something that Turbo should be able to handle for us.

Does this work instead? https://stackoverflow.com/a/76283733/421243

> 
> For GET forms and links you have to specify that you want to send a TURBO_STREAM request:
> ```
> <%= form_with url: search_orders_path, method: :get,
>    data: {turbo_stream: true} do |f| %>
> ...
> <% end %> 
> ```